content/intro-to-storybook/ember/en/screen.md

```diff
@@ -0,0 +1,201 @@
+      ),
+    };
+  };
+}
+
+// The reducer describes how the contents of the store change for each action
+export const reducer = (state, action) => {
+  switch (action.type) {
+    case actions.ARCHIVE_TASK:
+      return taskStateReducer('TASK_ARCHIVED')(state, action);
+    case actions.PIN_TASK:
+      return taskStateReducer('TASK_PINNED')(state, action);
+    case actions.SET_ERROR:
+      return {
+        ...state,
+        isError: true,
+      };
+    default:
+      return state || initialState;
+  }
+};
+
+export default combineReducers({
+  reducer,
+});
+```
+
+The store is updated with the new field. Let's create a presentational `PureInboxScreen.hbs` in our `app/components` folder:
+
+```handlebars
+{{!--app/components/PureInboxScreen.hbs--}}
+{{#if @error}}
+  <div class="page lists-show">
+    <div class="wrapper-message">
+        <span class="icon-face-sad" />
+        <div class="title-message">Oh no!</div>
+        <div class="subtitle-message">Something went wrong</div>
+    </div>
+  </div>
+{{else}}
+  <div class="page lists-show">
+    <nav>
+        <h1 class="title-page">
+            <span class="title-wrapper">Taskbox</span>
+        </h1>
+    </nav>
+    {{TaskList}}
+  </div>
+{{/if}}
+```
+
+Then, we can create a container, which again grabs the data for the `PureInboxScreen` in `app/components/InboxScreen.hbs` add the following:
+
+```handlebars
+{{!--app/components/InboxScreen.hbs--}}
+<div>
+    {{PureInboxScreen error=error}}
+</div>
+```
+
+In `app/components/InboxScreen.js` add the following:
+
+```javascript
+// app/components/InboxScreen.js
+import Component from '@glimmer/component';
+import { connect } from 'ember-redux';
+import { computed } from '@ember/object';
+
+const stateToComputed = state => {
+  const { reducer } = state;
+  const { isError } = reducer;
+
+  return {
+    error: isError,
+  };
+};
+class InboxScreen extends Component {
+  constructor() {
+    super(...arguments);
+  }
+  @computed('error')
+  get error() {
+    return this.error;
+  }
+}
+export default connect(stateToComputed)(InboxScreen);
+```
+
+We also change the `application` template to render the `InboxScreen` (eventually we would use a router to choose the correct screen, but let's not worry about that here):
+
+```handlebars
+{{!-- app/templates/aplication.hbs --}}
+
+{{InboxScreen}}
+```
+
+However, where things get interesting is in rendering the story in Storybook.
+
+As we saw previously, the `TaskList` component is a **container** that renders the `PureTaskList` presentational component. By definition with other frameworks, container components cannot be simply rendered in isolation; they expect to be passed some context or to connect to a service.
+
+When placing the `TaskList` into Storybook, we were able to ilustrate this issue by simply rendering the `PureTaskList` and avoiding the container. We'll do something similar and render the `PureInboxScreen` in Storybook also.
```

Comment:
Isn't there a problem with not having a redux store available when rendering the version that tries to grab data?

Reply:
We kind of want there to be a problem here so we can demonstrate using decorators to mock out data sources. I'm not sure how that works in Ember but I think we are missing a chance to teach people here..

Reply:
@tmeasday this is a bit like what happens with svelte. In this case the app is built when the `start` script runs and put into the `dist` folder. All bundled up already and Storybook "piggy backs" (pardon the bad pun) on it and serves the content of the folder basically. When i was testing this, when i reached the data section i was thinking if this would work or how would i have to mock the store with a decorator and when i ran Storybook it was working and presented me the data and was triggering the actions.